main_exercise/client/src/Components/TodoList/TodoList.js

```diff
@@ -0,0 +1,115 @@
+import ListHeader from '../listHeader/listHeader'
+import item_client from '../../clients/item_client'
+import { useEffect, useState } from 'react'
+import styles from './TodoList.module.css'
+import TodoItem from '../TodoItem/TodoItem'
+import Loader from '../Loader/Loader'
+
+export default function TodoList() {
+  const itemClient = new item_client()
+  const selectedArray = []
+  const [todos, setTodos] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+
+  async function updateStatus(id, status) {
+    try {
+      await itemClient.updateStatus(id, status)
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  async function updateUrgency(id, urgency) {
+    try {
+      await itemClient.updateUrgency(id, urgency)
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  async function initialTodos() {
+    try {
+      setIsLoading(true)
+      const todos = await itemClient.getTodos()
+      setTodos(todos)
+      setIsLoading(false)
```

Comment:
why do you need this twice `setIsLoading(false)`?
you can use `finally` and put it there..